core/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -364,7 +365,7 @@ private boolean addFailureIfIndexIsUnavailable(DocumentRequest request, BulkRequ
                 concreteIndex = concreteIndices.resolveIfAbsent(request.index(), request.indicesOptions());
             } catch (IndexClosedException ice) {
                 unavailableException = ice;
-            } catch (IndexMissingException ime) {
+            } catch (ResourceNotFoundException ime) {
```

Comment:
I'm confused - shouldn't this be IndexNotFoundException which seems to be thrown?